### PR TITLE
Feature/battery cell model

### PR DESCRIPTION
**Pull request by ZackTully**
This is a more detailed battery model called `LIB` based on the lithium ion cell model in [1.]. The `LIB` model calculates charging/discharging losses due to internal resistance and transient behavior.

This PR also includes initial documentation for the battery model #68.

### References:

1. M.-K. Tran et al., “A comprehensive equivalent circuit model for lithium-ion batteries, incorporating the effects of state of health, state of charge, and temperature on model parameters,” Journal of Energy Storage, vol. 43, p. 103252, Nov. 2021, doi: 10.1016/j.est.2021.103252.

### CODE REVIEW

File: hercules/emulator.py
Comment (misi9170) [inline hercules/emulator.py:269]: Perhaps this line can simply be removed? It seems pretty redundant.

**Review by misi9170 — CHANGES_REQUESTED** [review]
This looks fine to me aside from a few small comments below. 

I did run into a bit of trouble when running the example though---I found that, in order to run the example, the file outputs/log_test_client.log had to _already_ exist in the example. I checked, and the same was true for the 02 example, so I don't think it's this PR that's causing the problem. @genevievestarke , have you run into this before? Can one of you confirm that if you delete the outputs/log_test_client.log file, the example does not run for you? If so, I'll open an issue.

File: tests/python_simulators_test/battery_test.py
Comment (ZackTully) [reply tests/python_simulators_test/battery_test.py:1]: Removed description completely.

File: example_case_folders/02_amr_wind_standin_only/loghelics
Comment (ZackTully) [reply example_case_folders/02_amr_wind_standin_only/loghelics:1]: Deleted `loghelics`.

File: hercules/emulator.py
Comment (ZackTully) [reply hercules/emulator.py:269]: I agree it probably doesn't need to be printed every time step. Deleted.

File: tests/python_simulators_test/battery_test.py
Comment (ZackTully) [reply tests/python_simulators_test/battery_test.py:215]: Changed to `assert_array_almost_equal` up to 9 decimal points. If that still fails, I can think about a different way to test this behavior.

**Comment by ZackTully** [comment]
@misi9170 I tried running both example 02 and 06 and they seem to work _mostly_ fine for me.

I tried:
1. example 2 with whole `outputs` folder deleted.
2. example 2 with only `log_test_client.log` deleted.
3. example 6 with `outputs` folder deleted.
4. example 6 with only `log_test_client.log` deleted.

Case 1 was the only one that was unusual. When I ran the bash script, it would create the outputs folder then get stuck and not run the simulation. If I ran the bash script a second time, the outputs folder would be populated with the appropriate files.

**Comment by misi9170** [comment]
> @misi9170 I tried running both example 02 and 06 and they seem to work _mostly_ fine for me.
> 
> I tried:
> 
>     1. example 2 with whole `outputs` folder deleted.
> 
>     2. example 2 with only `log_test_client.log` deleted.
> 
>     3. example 6 with `outputs` folder deleted.
> 
>     4. example 6 with only `log_test_client.log` deleted.
> 
> 
> Case 1 was the only one that was unusual. When I ran the bash script, it would create the outputs folder then get stuck and not run the simulation. If I ran the bash script a second time, the outputs folder would be populated with the appropriate files.

Thanks @ZackTully ! That's consistent with what I was getting, exact that for me, it seems that I was getting the same problem you describe for Case 1 when trying Case 3. Anyway, I think that #93 should fix this; we'll hopefully get that one in soon and that should clear the way for this PR to go in

**Review by misi9170 — APPROVED** [review]
Approving now, but I think we should wait until #93 goes in before merging

**Comment by genevievestarke** [comment]
Which bash script are we using for this? The file 'batch_script.sh' seems most complete, but I'm getting odd output from the amr wind standin data part that we might want to look into? It looks like the wind speed in amr_standin_data.csv is just half the time value.

**Comment by genevievestarke** [comment]
Nvm, I figured it out, all good!

**Comment by genevievestarke** [comment]
It doesn't seem like the battery is being used in this example, or the power outputs aren't being updated. I thought we had some basic signals passed from WHOC in this example? Is this something we need to work out about the communication between the two (Hercules and WHOC)?

**Comment by ZackTully** [comment]
> It doesn't seem like the battery is being used in this example, or the power outputs aren't being updated. I thought we had some basic signals passed from WHOC in this example? Is this something we need to work out about the communication between the two (Hercules and WHOC)?

Thanks for catching that! It's a result of updating WHOC/HERCULES communication when the `emulator` main dictionary wasn't writing to the output file. The changes to WHOC are in this pull request:  https://github.com/NREL/wind-hybrid-open-controller/pull/33

I think that should solve the problem.

**Review by genevievestarke — APPROVED** [review]
Ok, sounds good! I think we can merge this in!

**Pull request merged**
